unit_scaling/tests/test_analysis.py

```diff
@@ -0,0 +1,131 @@
+# Copyright (c) 2023 Graphcore Ltd. All rights reserved.
+
+import pandas as pd
+import torch.nn.functional as F
+from torch import Tensor, nn, randn
+
+from ..analysis import graph_to_dataframe, plot
+from ..transforms import track_scales
+
+
+def test_graph_to_dataframe() -> None:
+    class Model(nn.Module):
+        def __init__(self, dim: int) -> None:
+            super().__init__()
+            self.dim = dim
+            self.linear = nn.Linear(dim, dim // 2)
+
+        def forward(self, x: Tensor) -> Tensor:  # pragma: no covers
+            y = F.relu(x)
+            z = self.linear(y)
+            return z.sum()  # type: ignore[no-any-return]
+
+    b, dim = 2**4, 2**8
+    input = randn(b, dim, requires_grad=True)
```

Comment:
If you forget `requires_grad=True` on the input, is it reasonably graceful?

Reply:
I think that's grand, it is something I've already hit when playing with analysis briefly, so that behaviour sounds helpful.